### PR TITLE
testing/i3wm-gaps: new aport

### PR DESCRIPTION
**Pull request by skrzyp**
This PR adds a fork of `i3wm` with Airblader patches too large to fit into upstream.
While it's a very popular fork, I thought it would be nice to have it in Alpine too.

Works very well, `musl.patch` from regular `i3wm` is still needed and applies well.
I've also added `xcb-util-xrm` which is a required library for `i3wm-gaps` now.

### CODE REVIEW

File: testing/i3wm-gaps/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+url="https://github.com/Airblader/i3"
+arch="all"
+license="BSD"
+replaces="i3wm"
+makedepends="bison flex libxcb-dev xcb-util-cursor-dev
+	xcb-util-keysyms-dev xcb-util-wm-dev libev-dev pango-dev cairo-dev
+	yajl-dev startup-notification-dev pcre-dev libxkbcommon-dev
+	xcb-util-xrm-dev asciidoc perl xmlto autoconf automake"
+subpackages="$pkgname-doc"
+source="i3wm-$pkgver.tar.gz::https://github.com/Airblader/i3/archive/4.13.tar.gz
```
Comment (jirutka) [inline testing/i3wm-gaps/APKBUILD]: Please don’t hard-code version in source URL (it complicates bumping of abuild), use variable `$pkgver`.

File: testing/i3wm-gaps/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	musl.patch"
+
+builddir="$srcdir/i3-$pkgver"
+
+build() {
+	cd "$srcdir/i3-$pkgver"
+        autoreconf -fi
```
Comment (jirutka) [inline testing/i3wm-gaps/APKBUILD]: Unfortunately APKBUILD does not run with `set -e` (yet, it’s on our TODO list to change it), so every statement that may fail, but should not fail, and is not the last statement of the function, should end with `|| return 1`.

File: testing/xcb-util-xrm/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# Maintainer: Jakub Skrzypnik <j.skrzypnik@openmailbox.org>
+pkgname=xcb-util-xrm
+pkgver=1.2
+pkgrel=0
+pkgdesc="Utility functions for the X resource manager"
+url="https://github.com/Airblader/xcb-util-xrm/"
+arch="all"
+license="MIT"
+makedepends="libxcb-dev xcb-util-dev util-macros bsd-compat-headers automake autoconf libx11-dev"
+source="https://github.com/Airblader/xcb-util-xrm/releases/download/v1.2/xcb-util-xrm-1.2.tar.gz"
```
Comment (jirutka) [inline testing/xcb-util-xrm/APKBUILD]: Please don’t hard-code version in source URL (it complicates bumping of abuild), use variable `$pkgver`.

File: testing/xcb-util-xrm/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+subpackages="$pkgname-dev"
+
+builddir="$srcdir/$pkgname-$pkgver"
+build() {
+	cd "$builddir"
+	autoreconf -fi
```
Comment (jirutka) [inline testing/xcb-util-xrm/APKBUILD]: Unfortunately APKBUILD does not run with `set -e` (yet, it’s on our TODO list to change it), so every statement that may fail, but should not fail, and is not the last statement of the function, should end with `|| return 1`.

File: testing/i3wm-gaps/APKBUILD (hunk 1 above)
Comment (skrzyp) [inline testing/i3wm-gaps/APKBUILD]: sorry for that, forgot about `$pkgver`

File: testing/i3wm-gaps/APKBUILD (hunk 2 above)
Comment (skrzyp) [inline testing/i3wm-gaps/APKBUILD]: oh okay

**Comment by jirutka** [comment]
Build fails:

```
Can't exec "libtoolize": No such file or directory at /usr/share/autoconf/Autom4te/FileUtils.pm line 345, <GEN3> line 4.
autoreconf: failed to run libtoolize: No such file or directory
autoreconf: libtoolize is needed because this package uses Libtool
```

**Comment by skrzyp** [comment]
It passes the build now, could we finally merge that?

File: testing/xcb-util-xrm/APKBUILD [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# Maintainer: Jakub Skrzypnik <j.skrzypnik@openmailbox.org>
+pkgname=xcb-util-xrm
+pkgver=1.2
+pkgrel=0
+pkgdesc="Utility functions for the X resource manager"
+url="https://github.com/Airblader/xcb-util-xrm/"
+arch="all"
+license="MIT"
+makedepends="libx11-dev libxcb-dev xcb-util-dev util-macros bsd-compat-headers autoconf automake libtool"
+source="https://github.com/Airblader/xcb-util-xrm/releases/download/v1.2/xcb-util-xrm-$pkgver.tar.gz"
```
Comment (jirutka) [inline testing/xcb-util-xrm/APKBUILD]: Once again, please do not hard-code version number. ;)

```diff
- source="https://github.com/Airblader/xcb-util-xrm/releases/download/v1.2/xcb-util-xrm-$pkgver.tar.gz"
+ source="https://github.com/Airblader/xcb-util-xrm/releases/download/v$pkgver/xcb-util-xrm-$pkgver.tar.gz"
```

File: testing/xcb-util-xrm/APKBUILD [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+subpackages="$pkgname-dev"
+
+builddir="$srcdir/$pkgname-$pkgver"
+build() {
+	cd "$builddir"
+	autoreconf -fi || return 1
```
Comment (jirutka) [inline testing/xcb-util-xrm/APKBUILD]: Do you have some special reason why you run `autoreconf`? The source tarball already contains `./configure` script.

File: testing/i3wm-gaps/APKBUILD [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  install -d "$pkgdir/usr/share/man/man1"
+  install -m644 man/*.1 "$pkgdir"/usr/share/man/man1/ || return 1
+}
+
+md5sums="a591216c8a84d27431b057e4d01b151e  i3wm-4.13.tar.gz
+aa998beb0b9a83d2910508768ac7faae  musl.patch"
+sha256sums="e80c07e525a07f2d3d2f02d282774d3264b6d1f165713c9903c06626ba4d2571  i3wm-4.13.tar.gz
+b7d084c53addc71fda13ccb8b3dcca2a32d9ca1590a28bf3be1a0c61870d3817  musl.patch"
+sha512sums="9b0c888c1c412622282f5e56ba6763eba3af88682d1652d1a1e6ce4e7fcdbec30e4fd6731b7e4d306bd33871a2581e0979714b47c9614210302c491ce9366ba8  i3wm-4.13.tar.gz
+8ce7d00371c43b93dabbe0dadf9caf7c58a68f4a0079f5a9b9552c15c55bfa0df16d7e87a281595af2ac5254632ba28ccf82a467cea16159b41490f6f2910299  musl.patch"
```
Comment (jirutka) [inline testing/i3wm-gaps/APKBUILD]: Please use only tabs for indentation (and alignment) in APKBUILDs and other shell scripts.

**Comment by fabled** [comment]
This package was merged from another PR earlier. Please submit new PR if any changes are required.

**Pull request closed without merging**